cmd/clusterctl/client/cluster/client.go

```diff
@@ -21,14 +21,33 @@ import (
 	"time"
 
 	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/wait"
 
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/config"
 	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/repository"
 	yaml "sigs.k8s.io/cluster-api/cmd/clusterctl/client/yamlprocessor"
 	logf "sigs.k8s.io/cluster-api/cmd/clusterctl/log"
 )
 
+var (
+	// CurrentContractVersion return the contract version supported by this Cluster API version.
+	// Note: Each Cluster API version supports one contract version, and by convention the contract version matches the current API version.
+	CurrentContractVersion = clusterv1.GroupVersion.Version
+
+	// GetCompatibleContractVersions return the list of contract version compatible with a given contract version.
+	// NOTE: A contract version might be compatible with older contract versions e.g. to allow users time to transition to the new API.
+	// NOTE: The return value must include also the contract version received in input.
+	GetCompatibleContractVersions = func(contract string) sets.Set[string] {
+		compatibleContracts := sets.New(contract)
+		if contract == "v1beta2" {
+			compatibleContracts.Insert("v1beta1")
+		}
+		return compatibleContracts
+	}
```

Comment:
This is the only function that we should change in future to drop compatibility with the v1beta1 contract.

Note: Test functions are not using "official" `CurrentContractVersion` and `GetCompatibleContractVersions` but a copy of those so we can keep testing support for compatible contracts also in case there are no compatible contracts for a given CAPI version.
e.g. See variables at the beginning of [cmd/clusterctl/client/cluster/installer_test.go](https://github.com/kubernetes-sigs/cluster-api/pull/12018/files#diff-2e4b988d0d6560a8808bf310390144171be52ed25dba0fa905ec4b5dd90b8c05)